### PR TITLE
support _tidb_rowid for table scan range

### PR DESCRIPTION
**Pull request by yu34po**
### What problem does this PR solve? 
fix https://github.com/pingcap/tidb/issues/7946

### What is changed and how it works?
when table without primary key, check the  last column of datasource is ExtraHandleCol,
then  use  ExtraHandleCol to calcute path ranges. 
count of tablescan is still incorrect

### Check List 

Tests 

 - Integration test

Related changes

 - Need to be included in the release note

### CODE REVIEW

**Comment by ngaut** [comment]
Coool

**Comment by jackysp** [comment]
Thanks @yu34po!
PTAL @winoros .

File: planner/core/logical_plans.go [hunk 1]
```diff
@@ -340,7 +340,16 @@ func (ds *DataSource) deriveTablePathStats(path *accessPath) (bool, error) {
 	}
 	if pkCol == nil {
 		path.ranges = ranger.FullIntRange(false)
-		return false, nil
+		if len(ds.Columns) > 0 {
+			extraHandleCol := ds.Columns[len(ds.Columns)-1]
+			if extraHandleCol.ID == model.ExtraHandleID {
+				pkCol = expression.ColInfo2Col(ds.schema.Columns, extraHandleCol)
```
Comment (winoros) [inline planner/core/logical_plans.go]: `DataSource.Columns[i]` matches `DataSource.Schema.Columns[i]`.

File: cmd/explaintest/t/explain_easy.test [hunk 2]
```diff
@@ -93,3 +93,10 @@ explain select ifnull(nb, 0), ifnull(nb, 0) from t;
 explain select 1+ifnull(nb, 0) from t;
 explain select 1+ifnull(a, 0) from t;
 drop table if exists t;
+drop table if exists t;
+create table t(a int);
+explain select * from t where _tidb_rowid = 0;
+explain select * from t where _tidb_rowid > 0;
+explain select a, _tidb_rowid from t where a > 0;
```
[the remark this answers is not in the document]
Comment (yu34po) [reply cmd/explaintest/t/explain_easy.test]: execute "explain select * from t where _tidb_rowid > 0;"
should get
"Error 1054: Unknown column '_tidb_rowid' in 'where clause'"
cannot add this test case into explaintest

**Comment by winoros** [comment]
@yu34po You can use `fix issue link` grammar. See https://help.github.com/articles/closing-issues-using-keywords/

**Pull request merged**